MergerCli/Program.cs

```diff
@@ -81,31 +83,31 @@ private static void Main(string[] args)
                         stopWatch.Reset();
                         stopWatch.Start();
 
-                        Console.WriteLine("Validating merged data sources");
-                        proccess.Validate(baseData, sources[i]);
+                        logger.LogInformation("Validating merged data sources");
+                        process.Validate(baseData, sources[i]);
 
                         stopWatch.Stop();
                         // Get the elapsed time as a TimeSpan value.
                         ts = stopWatch.Elapsed;
-                        timeUtils.PrintElapsedTime($"{sources[i].Path} validation time", ts);
+                        logger.LogInformation(timeUtils.FormatElapsedTime($"{sources[i].Path} validation time", ts));
                     }
                 }
             }
             catch (Exception ex)
             {
                 //save status on unhandled exceptions
                 OnFailure();
-                Console.WriteLine(ex.ToString());
+                logger.LogError(ex.ToString());
```

Comment:
doesn't the logger suppose to get exception as an argument

Reply:
yes but it requires additional parameters (such as message)

Reply:
add the message "failed to ..."

Reply:
this is for unhandled exceptions. it can be result of any thing I don't know what it failed to do....